components/ingest-service/server/jobs.go

```diff
@@ -1,124 +1,256 @@
+	err := c.RegisterTaskExecutor("delete_expired", &DeleteExpiredMarkedNodesTask{client}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+	err = c.RegisterTaskExecutor("missing_nodes", &MarkNodesMissingTask{client}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+	err = c.RegisterTaskExecutor("missing_nodes_for_deletion", &MarkMissingNodesForDeletionTask{client}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+	err = c.RegisterWorkflowExecutor("delete_expired", NewSingleTaskWorkflow("delete_expired"))
+	if err != nil {
+		return err
+	}
+	err = c.RegisterWorkflowExecutor("missing_nodes", NewSingleTaskWorkflow("missing_nodes"))
+	if err != nil {
+		return err
+	}
+	return c.RegisterWorkflowExecutor("missing_nodes_for_deletion", NewSingleTaskWorkflow("missing_nodes_for_deletion"))
+}
+
+func MigrateJobsSchedule(c *cereal.Manager, oldConfigFile string) error {
+	jc, err := config.OldConfigOrDefault(oldConfigFile)
+	if err != nil {
+		return errors.Wrap(err, "failed to read old job config from disk")
+	}
 
-	switch jc.ID {
-	case config.DeleteNodes:
-		cronFunc = func() {
-			deleteExpiredMarkedNodes(context.Background(), jc, server.client)
+	for idx, name := range config.JobList {
+		scheduleName := fmt.Sprintf("periodic_%s", name)
+		config, err := jc.ConfigForJob(idx)
+		if err != nil {
+			return err
 		}
-	case config.NodesMissing:
-		cronFunc = func() {
-			markNodesMissing(context.Background(), jc, server.client)
+
+		d, err := time.ParseDuration(config.Every)
+		if err != nil {
+			return errors.Wrap(err, "could not parse job configuration")
+		}
+		r, err := rrule.NewRRule(rrule.ROption{
+			Freq:     rrule.SECONDLY,
+			Interval: int(d.Seconds()),
+			Dtstart:  time.Now(),
+		})
+		if err != nil {
+			return errors.Wrap(err, "could not create recurrence rule for job configuration")
 		}
-	case config.MissingNodesForDeletion:
-		cronFunc = func() {
-			markMissingNodesForDeletion(context.Background(), jc, server.client)
+
+		err = c.CreateWorkflowSchedule(scheduleName, name, config.Threshold, config.Running, r)
+		if err == cereal.ErrWorkflowScheduleExists {
+			logrus.Infof("Schedule for %s already exists, not migrating", scheduleName)
+		} else if err != nil {
+			return errors.Wrap(err, "could not create scheduled workflow")
 		}
-	default:
-		log.Errorf("Job ID %d is not supported", jc.ID)
-		return
 	}
+	return nil
+}
+
+// TODO(ssd) 2019-05-15: This is a helper to avoid having to write
+// workflows for things that are just single tasks. Perhaps the
+// workflow library could have a helper that..helps with this.
+type SingleTaskWorkflow struct {
+	taskName string
+}
+
+func NewSingleTaskWorkflow(taskName string) *SingleTaskWorkflow {
+	return &SingleTaskWorkflow{taskName}
+}
+
+func (s *SingleTaskWorkflow) OnStart(w cereal.WorkflowInstance, ev cereal.StartEvent) cereal.Decision {
+	var params string
+	err := w.GetParameters(&params)
```

Comment:
wtf does the json parser do when you try unmarshal a struct to a string?

Reply:
We aren't doing that.  We are unmarshalling a string into a string.  A string is valid json. 

https://play.golang.org/p/1Ti7P1AMmEG

That said, this will probably change to a struct for some other reasons in my next push.

Reply:
Yeah, your explanation of it being a straight up byte array of some go type makes sense.